src/content/peerconnection/endtoend-encryption/js/main.js

```diff
@@ -79,6 +75,41 @@ function start() {
       });
 }
 
+// We use a Worker to do the encryption and decryption.
+// See
+//   https://developer.mozilla.org/en-US/docs/Web/API/Worker
+// for basic concepts.
+const worker = new Worker('./js/worker.js');
+function setupSenderTransform(sender) {
+  const senderStreams = sender.track.kind === 'video' ? sender.createEncodedVideoStreams() : sender.createEncodedAudioStreams();
+  // Instead of creating the transform stream here, we do a postMessage to the worker. The first
+  // argument is an object defined by us, the sceond a list of variables that will be transferred to
+  // the worker. See
+  //   https://developer.mozilla.org/en-US/docs/Web/API/Worker/postMessage
+  /*
+  const transformStream = new TransformStream({
+    transform: encodeFunction,
+  });
+  senderStreams.readableStream
+      .pipeThrough(transformStream)
+      .pipeTo(senderStreams.writableStream);
+  */
+  worker.postMessage({
+    type: 'encode',
+    readableStream: senderStreams.readableStream,
+    writableStream: senderStreams.writableStream,
+  }, [senderStreams.readableStream, senderStreams.writableStream]);
+}
+
+function setupReceiverTransform(receiver) {
+  const receiverStreams = receiver.track.kind === 'video' ? receiver.createEncodedVideoStreams() : receiver.createEncodedAudioStreams();
+  worker.postMessage({
+    type: 'decode',
+    readableStream: receiverStreams.readableStream,
```

Comment:
Suggest having a field called "operation" with the value "decode" and a field "type" with the value receiver.track.kind.

Reply:
why do we need to have the kind? (I think this relates to the other discussion we're having about having the kind on the frame)

Reply:
you asked for it.... when complaining about having to use "undefined" as a key value when determining the kind of the track.

Reply:
right. but if you look at the current code we're not creating a bound audio transform or video transform. We could but I don't think this will be "normal" way to use it. We need more opinions of people using this :-)